.github/actions/setup-java-maven/action.yml

```diff
@@ -0,0 +1,42 @@
+name: Setup Java and Maven
+description: |
+  This action setup Java and Maven and optionally caches the build artifacts.
+
+inputs:
+  JAVA_DISTRIBUTION:
+    description: |
+      The distribution of the JDK to use.
+    required: false
+    default: 'zulu'
+  JAVA_VERSION:
+    description: |
+      The version of the JDK to use.
+    required: false
+    default: '17'
+  JAVA_CACHE:
+    description: |
+      The name of the Java cache to use.
+    required: false
+  JAVA_OVERRIDE_SETTINGS:
```

Comment:
Rename this to MAVEN_OVERRIDE_SETTINGS.
While it is consumed by the setup java action it is actually used to control Maven behaviour.

Reply:
i am against this naming, as the current on is action _ option and then we could discuss much more naming